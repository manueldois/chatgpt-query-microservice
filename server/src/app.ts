console.log('\n ---- STARTING SERVER ---- \n')