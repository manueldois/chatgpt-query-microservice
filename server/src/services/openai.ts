import { Configuration, OpenAIApi } from "openai";
import fs from 'fs';

const openAiConfig = new Configuration({
    apiKey: fs.readFileSync(process.env.OPENAI_API_KEY_FILE).toString(),
});

export const openai = new OpenAIApi(openAiConfig);