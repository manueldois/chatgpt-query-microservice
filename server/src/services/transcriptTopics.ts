import crypto from 'crypto'

export const hashTranscript = (transcript: string) => crypto.createHash('md5').update(transcript).digest("hex")
