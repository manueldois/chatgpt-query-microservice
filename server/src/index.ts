import 'dotenv/config'
import path from "path";
import fs from 'fs'
import { createServer } from "./server";
import { Sequelize } from "sequelize";
import { Configuration, OpenAIApi } from 'openai'

const sequelize = new Sequelize({
    dialect: 'sqlite',
    storage: path.join(__dirname, '../../data/db.sqlite'),
    logging: false,
});

const openAiConfig = new Configuration({
    apiKey: fs.readFileSync(process.env.OPENAI_API_KEY_FILE).toString(),
});

const openai = new OpenAIApi(openAiConfig);

createServer({ sequelize, openai })
