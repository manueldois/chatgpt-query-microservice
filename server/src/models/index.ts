import { Sequelize } from "sequelize";
import path from 'path'
import { InitTranscriptTopics } from "./transcriptTopics";

export const sequelize = new Sequelize({
    dialect: 'sqlite',
    storage: path.join(__dirname, '../../db.sqlite'),
    logging: false,
});

InitTranscriptTopics(sequelize)
