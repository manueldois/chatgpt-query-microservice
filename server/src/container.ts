import fs from 'fs';
import path from "path";
import { createContainer, InjectionMode, asFunction, AwilixContainer } from "awilix";
import { Configuration, OpenAIApi } from "openai";
import { Model, ModelStatic, Sequelize } from "sequelize";
import { createTranscriptTopics } from './models/transcriptTopics';

export interface Cradle {
    sequelize: Sequelize,
    openai: OpenAIApi,
    TranscriptTopics: ModelStatic<Model<any, any>>
}

export let container = createContainer<Cradle>({
    injectionMode: InjectionMode.CLASSIC
})

export function setContainer(_container: AwilixContainer<Cradle>) {
    container = _container
}

export type DependenciesContainer = typeof container

container.register({
    sequelize: asFunction(createSequelize).singleton(),
    openai: asFunction(createOpenAiAPI).singleton(),
    TranscriptTopics: asFunction(createTranscriptTopics).singleton()
})

function createSequelize() {
    return new Sequelize({
        dialect: 'sqlite',
        storage: path.join(__dirname, '../data/db.sqlite'),
        logging: false,
    });
}

function createOpenAiAPI() {
    const openAiConfig = new Configuration({
        apiKey: fs.readFileSync(process.env.OPENAI_API_KEY_FILE).toString(),
    });

    return new OpenAIApi(openAiConfig);
}