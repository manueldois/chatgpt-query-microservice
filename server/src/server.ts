import 'dotenv/config'
import { createApp } from "./app";
import { DependenciesContainer } from './container';

export async function createServer(injectedContainer?: DependenciesContainer) {
    const app = await createApp(injectedContainer)

    return app.listen(
        3000,
        () => {
            console.log(`Server listening on port 3000`)
        }
    )
}

createServer()