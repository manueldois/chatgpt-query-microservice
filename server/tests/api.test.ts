import 'dotenv/config'
import { describe, it, vi } from 'vitest';
import request from 'supertest'
import { createApp } from '../src/app'
import { sequelize } from '../src/services/sequelize';

vi.mock('../src/services/sequelize', async () => {
  const { Sequelize } = await import('sequelize');
  return { sequelize: new Sequelize('sqlite::memory:') }
})

const app = await createApp()

describe('API tests', () => {
  it('Gets transcripts', async () => {

    await request(app).post('/transcripts').send({ transcript: "This is a text about apples" })

    const res = await request(app).get('/transcripts')

    console.log(res.body)
  })
})