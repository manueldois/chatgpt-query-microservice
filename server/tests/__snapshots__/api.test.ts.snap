// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`API tests > snapshot 1`] = `
{
  "foo": "bar",
}
`;
